File: benchmark/2026-09-17-7677581(latest)/standard/rubrics/probes/DBG-1/go/dbg1.go
// DBG-1
package main

import (
	"flag"
	"fmt"
	"os"
	"runtime/pprof"
)

const ITERATIONS int64 = 300000000
const MODULUS int64 = 1000000007

type Item struct {
	Count int64
	Name  string
}

func accumulate(items []Item, iterations int64) int64 {
	var total int64 = 0
	for i := int64(0); i < iterations; i++ {
		for _, item := range items {
			total = (total*31 + item.Count + int64(len(item.Name))) % MODULUS
		}
	}
	return total
}

var cpuprofile = flag.String("cpuprofile", "", "write cpu profile to file")

func main() {
	flag.Parse()
	if *cpuprofile != "" {
		f, err := os.Create(*cpuprofile)
		if err != nil {
			panic(err)
		}
		pprof.StartCPUProfile(f)
		defer pprof.StopCPUProfile()
	}
	items := []Item{}
	items = append(items, Item{Count: 7, Name: "alpha"})
	items = append(items, Item{Count: 11, Name: "bravo"})
	items = append(items, Item{Count: 13, Name: "charlie"})
	checksum := accumulate(items, ITERATIONS)
	fmt.Printf("checksum=%d\n", checksum)
}

File: benchmark/2026-09-17-7677581(latest)/standard/rubrics/probes/DBG-1/swift/dbg1.swift
// DBG-1
struct Item {
    var count: Int
    var name: String
}

let ITERATIONS: Int = 300000000
let MODULUS: Int = 1000000007

func accumulate(_ items: [Item], _ iterations: Int) -> Int {
    var total: Int = 0
    for _ in 0..<iterations {
        for item in items {
            total = (total * 31 + item.count + item.name.count) % MODULUS
        }
    }
    return total
}

var items: [Item] = []
items.append(Item(count: 7, name: "alpha"))
items.append(Item(count: 11, name: "bravo"))
items.append(Item(count: 13, name: "charlie"))
let checksum = accumulate(items, ITERATIONS)
print("checksum=\(checksum)")
